app_dart/lib/src/request_handlers/push_gold_status_to_github.dart

```diff
@@ -90,53 +74,80 @@ class PushGoldStatusToGithub extends ApiRequestHandler<Body> {
         continue;
       }
 
-      if (cirrusCheckStatuses.any(kCirrusInProgressStates.contains)) {
-        // Checks are still running, we have to wait.
-        log.debug('Waiting for checks to be completed.');
-        statusRequest = _createStatus(GithubGoldStatusUpdate.statusRunning,
-            'This check is waiting for all other checks to be completed.');
-      } else {
-        // Get Gold status.
-        final String goldStatus = await _getGoldStatus(pr, log);
-        statusRequest =
-            _createStatus(goldStatus, _getStatusDescription(goldStatus));
-        if (goldStatus == GithubGoldStatusUpdate.statusRunning &&
-            !await _alreadyCommented(gitHubClient, pr, slug)) {
-          log.debug('Notifying for triage.');
-          await _commentAndApplyGoldLabel(gitHubClient, pr, slug);
+      log.debug('Querying Cirrus for pull request #${pr.number}...');
+      cirrusCheckStatuses.clear();
+      bool runsGoldenFileTests = false;
+      // Query current checks for this pr.
+      final List<dynamic> cirrusChecks =
+          await queryCirrusGraphQL(pr.head.sha, cirrusClient, log, 'flutter');
+      for (dynamic check in cirrusChecks) {
+        final String status = check['status'];
+        final String taskName = check['name'];
```

Comment:
This is reminding me that we should turn off `avoid_as` and set `implicit-casts: false`.

I was going to suggest some ways to avoid dynamic typing here but we can save that for a PR that just changes those lints around.